content/blog/2021-02-17-cml-runner-prerelease.md

```diff
@@ -0,0 +1,165 @@
+---
+title: 'Pre-Release Notes: Automatically Train Models in the Cloud with CML'
+date: 2021-02-17
+description: |
+  New features are here to make launching cloud 
+  compute for continuous integration workflows 
+  shorter, sweeter and easier than ever. Plus, 
+  a new GitHub Action to setup CML means more 
+  ways to use CML without our Docker container.
+descriptionLong: |
+  New features are here to make launching cloud 
+  compute for continuous integration workflows 
+  shorter, sweeter and easier than ever. Plus, 
+  a new GitHub Action to setup CML means more 
+  ways to use CML without our Docker container.
+picture: 2021-02-17/gh_actions_art.png
+author: elle_obrien
+commentsUrl: https://discuss.dvc.org/t/cml-self-hosted-runners-on-demand-with-gpus/462
+tags:
+  - CML
+  - Continuous Machine Learning
+  - GitHub Actions
+  - GitLab
+  - Cloud
+---
+
+Today, we're pre-releasing some new features in Continuous Machine Learning, or
+[CML](https://cml.dev)- our open source project to adapt popular continuous
+integration (CI) systems like GitHub Actions and GitLab CI for data science.
+
+Here's what you'll get:
+
+## Brand new method to provision cloud compute for your CI workflows
+
+In the initial CML release, we provided a method user Docker Machine to launch
+instances on AWS, Azure and GCP from your GitHub Actions and GitLab CI
+workflows. We've completely redone this recipe using Terraform and now it's
+shorter, sweeter, and more powerful than ever!
+
+Check out this example workflow to launch an EC2 instance from a GitHub Action
+workflow and then train a model.
+
+```yaml
+name: 'Train in the cloud'
+on: [push]
+
+jobs:
+  deploy-runner:
+    runs-on: [ubuntu-latest]
+    steps:
+      - uses: iterative/setup-cml@v1
+      - uses: actions/checkout@v2
+      - name: deploy
+        shell: bash
+        env:
+          repo_token: ${{ secrets.PERSONAL_ACCESS_TOKEN }}
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+        run: |
+          cml-runner \
+          --cloud aws \
+          --cloud-region us-west \
+          --cloud-type=t2.micro \
+          --labels=cml-runner
+  train-model:
+    needs: deploy-runner
+    runs-on: [self-hosted, cml-runner]
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.x'
+      - name: 'Train my model'
+        run: |
+          pip install -r requirements.txt
+          python train.py
+```
+
+If you use CML functions in the `train-model` step, you can go even further and
+get a closed loop- sending model training results from the EC2 instance to your
+pull request or merge request! For example, if we expand the `train-model` step
+to incorporate functions like `cml-publish` and `cml-send-comment`:
+
+```yaml
+train-model:
+  needs: deploy-runner
+  runs-on: [self-hosted, cml-runner]
+  container: docker://dvcorg/cml-py3
```

Comment:
We dont need this, we can now use the setup-python action with the latest change I did today.
You can use ```docker://dvcorg/cml```

Reply:
ok cool. so it'll be the `docker://dvcorg/cml` container, plus the setup python action.